sdks/python/apache_beam/runners/interactive/recording_manager_test.py

```diff
@@ -149,43 +152,37 @@ def test_read_n(self):
 
   def test_read_duration(self):
     """Test that the stream only reads a 'duration' of elements."""
+    def as_windowed_value(element):
+      return WindowedValueHolder(WindowedValue(element, 0, []))
 
     values = (FileRecordsBuilder(tag=self.cache_key)
               .advance_processing_time(1)
-              .add_element(element=0, event_time_secs=0)
+              .add_element(element=as_windowed_value(0), event_time_secs=0)
               .advance_processing_time(1)
-              .add_element(element=1, event_time_secs=1)
+              .add_element(element=as_windowed_value(1), event_time_secs=1)
               .advance_processing_time(1)
-              .add_element(element=2, event_time_secs=3)
+              .add_element(element=as_windowed_value(2), event_time_secs=3)
               .advance_processing_time(1)
-              .add_element(element=3, event_time_secs=4)
+              .add_element(element=as_windowed_value(3), event_time_secs=4)
               .advance_processing_time(1)
-              .add_element(element=4, event_time_secs=5)
+              .add_element(element=as_windowed_value(4), event_time_secs=5)
               .build()) # yapf: disable
 
+    values = [
+        v.recorded_event for v in values if isinstance(v, TestStreamFileRecord)
+    ]
+
     self.mock_result.set_state(PipelineState.DONE)
     self.cache.write(values, 'full', self.cache_key)
-    self.cache.save_pcoder(None, 'full', self.cache_key)
-
-    # The elements read from the cache are TestStreamFileRecord instances and
-    # have the underlying elements encoded. This method decodes the elements
-    # from the TestStreamFileRecord.
-    def get_elements(events):
-      coder = coders.FastPrimitivesCoder()
-      elements = []
-      for e in events:
-        if not isinstance(e, TestStreamFileRecord):
-          continue
-
-        if e.recorded_event.element_event:
-          elements += ([
-              coder.decode(el.encoded_element)
-              for el in e.recorded_event.element_event.elements
-          ])
-      return elements
+    self.cache.save_pcoder(coders.FastPrimitivesCoder(), 'full', self.cache_key)
 
     # The following tests a progression of reading different durations from the
     # cache.
+
+    # Small convienence function for getting the values.
+    def get_elements(events):
+      return [e.value for e in events]
```

Comment:
I am trying to figure out - where did we stop passing encoded data events?

Reply:
The test uses the `InMemoryCache` set in the `setUp` method that directly writes values to an in-memory map verbatim.